decoder.go

```diff
@@ -57,17 +57,53 @@ func decodeRuleIdArray(obj *wafObject) ([]string, error) {
 	return ruleIds, nil
 }
 
-func decodeActions(cActions uintptr, size uint64) []string {
-	if size == 0 {
-		return nil
+func decodeObject(obj *wafObject) (interface{}, error) {
+	switch obj._type {
+	case wafMapType:
+		return decodeMap(obj)
+	case wafArrayType:
+		return decodeArray(obj)
+	case wafStringType:
+		return gostringSized(cast[byte](obj.value), obj.nbEntries), nil
+	default:
+		return nil, errUnsupportedValue
+	}
+}
+
+func decodeArray(obj *wafObject) ([]interface{}, error) {
+	if obj._type != wafArrayType {
+		return nil, errInvalidObjectType
+	}
+
+	var events []interface{}
```

Comment:
In this case we know how much capacity is required upfront, so if we `make` the map we might safe some reallocs.

```suggestion
	events := make([]any, obj.nbEntries)
```

Reply:
[0e8cf0a](https://github.com/DataDog/go-libddwaf/pull/30/commits/0e8cf0ae29eee053563ea20a196700e98b9eb83f)